src/libs/browser-storage.js

```diff
@@ -31,12 +28,74 @@ export const setDynamic = (storage, key, value) => {
   }
 }
 
-export const removeDynamic = (storage, key) => {
+export const getStatic = (storage, key) => {
+  return maybeParseJSON(storage.getItem(key))
+}
+
+export const setStatic = (storage, key, value) => {
+  if (value === undefined) {
+    storage.removeItem(key)
+  } else {
+    forceSetItem(storage, key, JSON.stringify(value))
+  }
+}
+
+export const listenStatic = (storage, key, fn) => {
+  window.addEventListener('storage', e => {
+    if (e.storageArea === storage && e.key === key) {
+      fn(maybeParseJSON(e.newValue))
+    }
+  })
+}
+
+export const getDynamic = (storage, key) => {
+  const storageKey = `dynamic-storage/${key}`
+  const data = maybeParseJSON(storage.getItem(storageKey))
+  return data && data.value
+}
+
+export const setDynamic = (storage, key, value) => {
+  const storageKey = `dynamic-storage/${key}`
+  if (value === undefined) {
+    storage.removeItem(storageKey)
+  } else {
+    forceSetItem(storage, storageKey, JSON.stringify({ timestamp: Date.now(), value }))
+  }
+}
+
+export const listenDynamic = (storage, key, fn) => {
   const storageKey = `dynamic-storage/${key}`
-  storage.removeItem(storageKey)
+  window.addEventListener('storage', e => {
+    if (e.storageArea === storage && e.key === storageKey) {
+      const data = maybeParseJSON(e.newValue)
+      fn(data && data.value)
+    }
+  })
 }
 
-const withUserPrefix = key => `${getUser().id}/${key}`
+/**
+ * Returns a stateful object that manages the given storage location.
+ * Implements an atom-compaible interface, and can be passed to useAtom.
+ */
+
+export const staticStorageSlot = (storage, key) => {
+  const { subscribe, next } = subscribable()
+  const get = () => getStatic(storage, key)
+  const set = newValue => {
+    setStatic(storage, key, newValue)
+    next(newValue)
+  }
+  listenStatic(storage, key, next)
+  return { subscribe, get, set, update: fn => set(fn(get())) }
+}
 
-export const getLocalPref = key => getDynamic(localStorage, withUserPrefix(key))
-export const setLocalPref = (key, value) => setDynamic(localStorage, withUserPrefix(key), value)
+export const dynamicStorageSlot = (storage, key) => {
```

Comment:
Since dynamic storage slots aren't currently used should we remove it for now, or do you envision this being used in the near future?

Reply:
That's a fair point, I don't think we have an immediate use case so perhaps we should remove it.